airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowTest.java

```diff
@@ -1013,6 +1028,185 @@ void testGetStuckAfterRun(Consumer<ConnectionManagerWorkflow> signalSender, Thre
 
   }
 
+  @Nested
+  @DisplayName("Test workflow where the child workflow throw a hearbeat timeout exception")
+  class HeartbeatFailureWorkflow {
+
+    @BeforeEach
+    public void setup() {
+      setupSpecificChildWorkflow(SyncWorkflowFailingWithHearbeatTimeoutException.class);
+    }
+
+    public static Stream<Arguments> getMaxAttemptForResetRetry() {
```

Comment:
can we make this global instead of scoped to the inner class? we could save some LOC

Reply:
Done